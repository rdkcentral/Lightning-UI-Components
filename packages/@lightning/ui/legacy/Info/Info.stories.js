import lng from '@lightningjs/core';
import Info from '.';
import mdx from './Info.mdx';

export default {
  title: 'Patterns / Info',
  parameters: {
    docs: { page: mdx }
  }
};

export const Basic = args =>
  class Basic extends lng.Component {
    static _template() {
      return {
        Info: {
          type: Info,
          title: args.title,
          subtitle: args.subtitle,
          description: args.description,
          announce: args.announce
        }
      };
    }

    _getFocused() {
      if (args.focused) {
        return this.tag('Info');
      }
    }
  };

Basic.args = {
  title: 'Series Title',
  subtitle: 'Season 5 - Episode 2',
  description:
    'This is a description of episode 2 from Season 5 of Series Title',
  announce: '',
  focused: false
};
Basic.argTypes = {
  focused: { control: 'boolean' }
};
Basic.parameters = {
  argActions: {
    focused: (isFocused, component) => {
      component._getFocused = isFocused
        ? () => component.tag('Info')
        : () => {};
      component._refocus();
    },
    title: (title, component) => {
      component.tag('Info').title = title;
    },
    subtitle: (subtitle, component) => {
      component.tag('Info').subtitle = subtitle;
    },
    description: (description, component) => {
      component.tag('Info').description = description;
    },
    announce: (announce, component) => {
      component.tag('Info').announce = announce;
    }
  }
};
