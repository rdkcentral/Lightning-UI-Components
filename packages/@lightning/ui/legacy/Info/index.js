import { Base, withStyles } from '@lightning/ui-core';
import styles from './Info.styles';
import DataItem from '../DataItem';

export default class Info extends withStyles(Base, styles) {
  static _template() {
    return {
      DataItem: {
        type: DataItem,
        h: 200,
        w: 410,
        margin: this.styles.margin.x,
        content: {
          flex: { direction: 'column' },
          w: w => w,
          Title: {
            text: {
              ...this.styles.title,
              wordWrapWidth: this.styles.wordWrapWidth
            }
          },
          Subtitle: {
            text: {
              ...this.styles.subtitle,
              wordWrapWidth: this.styles.wordWrapWidth
            }
          },
          Description: {
            text: {
              ...this.styles.description,
              wordWrapWidth: this.styles.wordWrapWidth
            }
          }
        }
      }
    };
  }

  static get properties() {
    return ['title', 'subtitle', 'description'];
  }

  static get tags() {
    return [
      'DataItem',
      { name: 'Content', path: 'DataItem.Content' },
      { name: 'Title', path: 'DataItem.Content.Title' },
      { name: 'Subtitle', path: 'DataItem.Content.Subtitle' },
      { name: 'Description', path: 'DataItem.Content.Description' }
    ];
  }

  _construct() {
    super._construct();
    this._paddingY = this.styles.padding.y;
  }

  _init() {
    this._Title.on('txLoaded', () => this._onTextLoaded());
    this._Subtitle.on('txLoaded', () => this._onTextLoaded());
    this._Description.on('txLoaded', () => this._onTextLoaded());
    this.requestUpdate(true);
  }

  _update() {
    this._updateText();
    this._updateFocusRing();
    this.fireAncestors('$itemChanged');
  }

  _updateText() {
    this._Title.text = this.title;
    this._Subtitle.text = this.subtitle;
    this._Description.text = this.description;
  }

  _onTextLoaded() {
    const contentHeight =
      this._Title.renderHeight +
      this._Subtitle.renderHeight +
      this._Description.renderHeight;

    const nextH = contentHeight + this._paddingY * 2;
    this._DataItem.h = nextH;
    this._DataItem._Background.h = nextH;
    this.requestUpdate();
  }

  _updateFocusRing() {
    if (this._DataItem._FocusRing) {
      this._DataItem._FocusRing.h = this._DataItem.h;
      this._DataItem._FocusRing.requestUpdate();
    }
  }

  _getFocused() {
    return this._DataItem;
  }
  set announce(announce) {
    super.announce = announce;
  }

  get announce() {
    if (this._announce) {
      return this._announce;
    }

    const description = this._Description.text.text
      ? [this._Description.text.text]
      : [];
    const subtitle = this._Subtitle.text.text ? [this._Subtitle.text.text] : [];
    const title = this._Title.text.text ? [this._Title.text.text] : [];
    return [...title, ...subtitle, ...description].join(',');
  }
}
