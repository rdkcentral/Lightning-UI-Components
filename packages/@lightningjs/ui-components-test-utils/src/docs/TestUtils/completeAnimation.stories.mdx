<!--
  Copyright 2023 Comcast Cable Communications Management, LLC

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.

  SPDX-License-Identifier: Apache-2.0
-->

import { Meta } from '@storybook/addon-docs';

<Meta title="Unit Testing / Test Utils / completeAnimation" />

# completeAnimation

Returns a Promise that resolves once all animating properties have updated to their target value(s).

## Arguments

`completeAnimation(element, transitionProperties = [])`

| argument             | type              | default | description                                                 |
| -------------------- | ----------------- | ------- | ----------------------------------------------------------- |
| element              | Lightning.Element | -       | element with properties that transition value changes       |
| transitionProperties | array\|string     | []      | property name(s) that transition value changes on component |

<br />

##### Example: Testing element properties before and after their values have transitioned

```js
import lng from '@lightningjs/core';
import { makeCreateComponent } from '@lightningjs/ui-components-test-utils';

class Example extends lng.Component {
  static _template() {
    return {
      TransitioningRect: {
        rect: true,
        color: 0xffffffff,
        w: 100,
        h: 100,
        x: 0,
        y: 0
      }
    };
  }

  _updatePosition() {
    this.tag('TransitioningRect').setSmooth('x', this.axisPosition);
    this.tag('TransitioningRect').setSmooth('y', this.axisPosition);
  }

  set axisPosition(value) {
    this._axisPosition = value;
    this._updatePosition();
  }

  get axisPosition() {
    return this._axisPosition;
  }
}

// generate a createComponent function to use in tests
const createExampleComponent = makeCreateComponent(Example);

// Ex. await the transitions of x and y values to finish before testing the result
it('should transition the x value of the component', async () => {
  const [example] = createExampleComponent();
  expect(example.tag('TransitioningRect').x).toBe(0);
  expect(example.tag('TransitioningRect').y).toBe(0);

  example.axisPosition = 50;
  await completeAnimation(example.tag('TransitioningRect'), ['x', 'y']);

  expect(example.tag('TransitioningRect').x).toBe(50);
  expect(example.tag('TransitioningRect').y).toBe(50);
});
```
