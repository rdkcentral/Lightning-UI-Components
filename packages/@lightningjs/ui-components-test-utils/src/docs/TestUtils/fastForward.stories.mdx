<!--
  Copyright 2023 Comcast Cable Communications Management, LLC

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.

  SPDX-License-Identifier: Apache-2.0
-->

import { Meta } from '@storybook/addon-docs';

<Meta title="Unit Testing / Test Utils / fastForward" />

# fastForward

Force all running Lightning Transitions on one or more Lightning Elements to finish and update the transitioning property to its target value immediately.

This function is similar to [completeAnimation](/story/unit-testing-test-utils-completeanimation--page) in that it allows testing transitioning properties. Unlike `completeAnimation`, `fastForward` is not asynchronous, and does result in all running transitions to complete. If a test requires awaiting specific properties to finish transitioning, `completeAnimation` is the recommended utility function. The `fastForward` function forces transitions to finish, but does not wait for the template to update before proceeding through the test function. To ensure the template has updated following a call to `fastForward`, invoke [`testRenderer.update()`](/story/unit-testing-test-renderer-overview--page).
// TODO: add link directly to testRenderer.update() docs once those are written

## Arguments

`fastForward(elements)`

| argument | type                                           | default | description                                   |
| -------- | ---------------------------------------------- | ------- | --------------------------------------------- |
| elements | Array<Lightning.Element\> \| Lightning.Element | -       | element(s) with transitioning property values |

<br />

##### Example: Forcing running Transitions to finish before testing the result

```js
import lng from '@lightningjs/core';
import {
  makeCreateComponent,
  fastForward
} from '@lightningjs/ui-components-test-utils';

class Example extends lng.Component {
  static _template() {
    const square = {
      rect: true,
      w: 100,
      h: 100,
      x: 0,
      y: 0,
      color: 0xeeeeeeee
    };
    return {
      TransitioningRectA: square,
      TransitioningRectB: square
    };
  }

  set axisPosition(value) {
    this._axisPosition = value;
    this.tag('TransitioningRectA').setSmooth('x', value);
    this.tag('TransitioningRectB').setSmooth('y', value);
  }

  get axisPosition() {
    return this._axisPosition;
  }
}

// generate a createComponent function to use in tests
const createExampleComponent = makeCreateComponent(Example);

// Ex. force all running transitions to complete, then test the result
it('should transition the x value of the component', () => {
  const [example, testRenderer] = createExampleComponent();
  expect(example.tag('TransitioningRectA').x).toBe(0);
  expect(example.tag('TransitioningRectB').y).toBe(0);

  //
  example.axisPosition = 50;
  const rectA = example.tag('TransitioningRectA');
  const rectB = example.tag('TransitioningRectB');
  fastForward([rectA, rectB]);
  testRenderer.update();

  expect(example.tag('TransitioningRectA').x).toBe(50);
  expect(example.tag('TransitioningRectB').y).toBe(50);
});
```
