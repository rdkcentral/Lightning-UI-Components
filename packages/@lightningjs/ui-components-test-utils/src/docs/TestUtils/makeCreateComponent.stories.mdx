<!--
  Copyright 2023 Comcast Cable Communications Management, LLC

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.

  SPDX-License-Identifier: Apache-2.0
-->

import { Meta } from '@storybook/addon-docs';

<Meta title="Docs / Unit Testing / Test Utils / makeCreateComponent " />

# makeCreateComponent

Returns a function, referred to as a `createComponent` function, which is used to generate an instance of a component to run unit test cases against.

## Arguments

`makeCreateComponent(component, defaultConfig, defaultOptions)`

| argument       | type          | default                                  | description                                                                                                                |
| -------------- | ------------- | ---------------------------------------- | -------------------------------------------------------------------------------------------------------------------------- |
| component      | lng.Component | the component to generate an instance of | component to render an instance of                                                                                         |
| defaultConfig  | object        | {}                                       | properties that should be applied to the component by default                                                              |
| defaultOptions | object        | {}                                       | default options used when creating the component instance (these options are passed to the `TestRenderer.create` function) |

### `createComponent`

Returns an array of 2 elements:

1. an instance of the component to be tested against (generated by `testRenderer.getInstance()`)
2. a [testRenderer](/story/unit-testing-test-renderer-create--page#testrenderer-functions) object generated using the `component` argument and a merged object created from `defaultOptions` and `options`

#### Arguments

| argument | type   | default | description                                                                                                                                                                |
| -------- | ------ | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| config   | object | {}      | properties that should be applied to the component. This object is spread over the `defaultConfig` object.                                                                 |
| options  | object | {}      | options used when creating the component instance (these options are passed to the `TestRenderer.create` function). This object is spread over the `defaultOptions` object |

<br />

##### Example: Generating a `makeComponent` function and test component

```js
import lng from '@lightningjs/core';
import { makeCreateComponent } from '@lightningjs/ui-components-test-utils';

// Lightning component to write unit tests for
class Example extends lng.Component {
  static _template() {
    return {
      Example: {
        text: { text: '' }
      }
    };
  }

  set exampleProp(value) {
    this._exampleProp = value;
    this.tag('Example').text.text = value;
  }

  get exampleProp() {
    return this._exampleProp;
  }
}

// unit tests for Example component

// generate a createComponent function to use in tests
const defaultConfig = { exampleProp: 'foo' };
const defaultOptions = { focused: false };
const createExampleComponent = makeCreateComponent(
  Example,
  defaultConfig,
  defaultOptions
);

// Ex. An instance of the component rendered with the properties defined in the defaultConfig and defaultOptions object
it('should render the exampleProp value as text', () => {
  const [example] = createExampleComponent();

  expect(example.exampleProp).toBe('foo');
  expect(example.tag('Example').text.text).toBe('foo');
  expect(example.hasFocus()).toBe(false);
});

// Ex. An instance of the component rendered with the properties defined in the config object
it('should support updating the text', () => {
  // overwrite the value of exampleProp defined in defaultConfig
  const [example] = createExampleComponent({
    exampleProp: 'bar'
  });

  expect(example.exampleProp).toBe('bar');
  expect(example.tag('Example').text.text).toBe('bar');
});

// Ex. An instance of the component rendered with options defined in the options object
it('should support rendering the component in a focused state', () => {
  // overwrite the value of focused defined in defaultOptions
  const [example] = createExampleComponent({}, { focused: true });

  expect(example.hasFocus()).toBe(true);
});

// Ex. Using a function provided by the testRenderer object (focus)
it('should allow focusing on the component', () => {
  const [example, testRenderer] = createExampleComponent();

  // focused was set to false in defaultOptions, so the componet will initially render without focus
  expect(example.hasFocus()).toBe(false);

  testRenderer.focus();

  expect(example.hasFocus()).toBe(true);
});
```

### spyOnMethods

A utility which enables running component methods asynchronously.

The `options` argument passed to the `createComponent` function accepts a field called `spyOnMethods`.
The value for `spyOnMethods` can be an array of method names to generate promises from. When one of those methods is invoked in the component it's associated promise will resolve, then the actual method will be invoked. The promises generated from the passed in methods are given the name `_${methodName}SpyPromise`.

<br />

##### Example: Spying on component methods and awaiting their execution before making assertions on their side effects

```js
class MethodSpiesExample extends Base {
  static get __componentName() {
    return 'MethodSpiesExample';
  }

  static _template() {
    return {
      Title: {
        type: TextBox
      },
      Subtitle: {
        type: TextBox,
        y: 100
      }
    };
  }

  static get tags() {
    return ['Title', 'Subtitle'];
  }

  static get properties() {
    return ['title', 'subtitle'];
  }

  _construct() {
    super._construct();
    this._title = 'title';
    this._subtitle = 'subtitle';
  }

  _update() {
    this._updateTitle();
    this._updateSubtitle();
  }

  _updateTitle() {
    this._Title.content = this.title;
  }
  _updateSubtitle() {
    this._Subtitle.content = this.subtitle;
  }
}

const createComponent = makeCreateComponent(MethodSpies);

it('should update the title and subtitle', async () => {
  const [methodSpiesComponent, testRenderer] = createComponent(
    {},
    { spyOnMethods: ['_update'] }
  );

  // wait for _update to be called on the initial render before proceeding
  await methodSpiesComponent.__updateSpyPromise;

  expect(methodSpiesComponent._Title.content).toBe('title');
  expect(methodSpiesComponent._Subtitle.content).toBe('subtitle');

  const newTitle = 'new title';
  const newSubtitle = 'new subtitle';
  methodSpiesComponent.patch({
    title: newTitle,
    subtitle: newSubtitle
  });

  // wait for _update to be called in response to values in the properties array changing
  await methodSpiesComponent.__updateSpyPromise;

  expect(methodSpiesComponent._Title.content).toBe(newTitle);
  expect(methodSpiesComponent._Subtitle.content).toBe(newSubtitle);
});
```

**Note: `spyOnMethods` can only be set in in the `options` object of `createComponent`. It can not be assigned to the `defaultOptions` object in `makeCreateComponent`**
