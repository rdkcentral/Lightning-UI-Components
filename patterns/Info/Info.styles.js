export default theme => ({
  title: theme.typography.headline1,
  subtitle: {
    ...theme.typography.button1,
    textColor: theme.palette.text.light.primary
  },
  description: {
    ...theme.typography.body3,
    textColor: theme.palette.text.light.secondary
  },
  margin: {
    x: theme.spacing(2)
  },
  padding: {
    y: theme.spacing(2)
  },
  wordWrapWidth: 410 - theme.spacing(4)
});
