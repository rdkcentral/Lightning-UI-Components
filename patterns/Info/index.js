import lng from '@lightningjs/core';
import DataItem from '../DataItem';
import withStyles from '../../mixins/withStyles';
import { getFocusScale } from '../../Styles';

export const styles = theme => ({
  title: theme.typography.headline1,
  subtitle: {
    ...theme.typography.button1,
    textColor: theme.palette.text.light.primary
  },
  description: {
    ...theme.typography.body3,
    textColor: theme.palette.text.light.secondary
  },
  margin: {
    x: theme.spacing(2)
  },
  padding: {
    y: theme.spacing(2)
  },
  wordWrapWidth: 410 - theme.spacing(4)
});

export default class Info extends withStyles(lng.Component, styles) {
  static _template() {
    return {
      DataItem: {
        type: DataItem,
        h: 200,
        w: 410,
        margin: this.styles.margin.x,
        content: {
          flex: { direction: 'column' },
          w: w => w,
          Title: {
            text: {
              ...this.styles.title,
              wordWrapWidth: this.styles.wordWrapWidth
            }
          },
          Subtitle: {
            text: {
              ...this.styles.subtitle,
              wordWrapWidth: this.styles.wordWrapWidth
            }
          },
          Description: {
            text: {
              ...this.styles.description,
              wordWrapWidth: this.styles.wordWrapWidth
            }
          }
        }
      }
    };
  }

  _construct() {
    super._construct();
    this.paddingY = this.styles.padding.y;
  }

  _update() {
    const contentHeight =
      this._Title.renderHeight +
      this._Subtitle.renderHeight +
      this._Description.renderHeight;
    const h = this.hasFocus()
      ? contentHeight * getFocusScale(this._DataItem.w)
      : contentHeight + this.paddingY * 2;
    this.h = h;
    this._DataItem.h = h;
    this._DataItem._Background.h = h;
    this._updateFocusRing();
    this.fireAncestors('$itemChanged');
  }

  _focus() {
    this.h =
      this._DataItem.h * getFocusScale(this._DataItem.w) + this._DataItem.y;
    this.fireAncestors('$itemChanged');
  }

  _unfocus() {
    this.h = this._DataItem.h + this._DataItem.y;
    this.fireAncestors('$itemChanged');
  }

  _getFocused() {
    return this._DataItem;
  }

  _loadText(comp, text) {
    if (!comp._hasTxLoader) {
      comp._hasTxLoader = true;
      comp.on('txLoaded', () => this._update());
    }
    comp.text = text;
  }

  _updateFocusRing() {
    if (this._DataItem._FocusRing) {
      this._DataItem._FocusRing.h = this.h;
      this._DataItem._FocusRing._update();
    }
  }

  set title(title) {
    if (this._Title.text.text !== title) {
      this._loadText(this._Title, title);
    }
  }

  set subtitle(subtitle) {
    if (this._Subtitle.text.text !== subtitle) {
      this._loadText(this._Subtitle, subtitle);
    }
  }

  set description(description) {
    if (this._Description.text.text !== description) {
      this._loadText(this._Description, description);
    }
  }

  get announce() {
    const description = this._Description.text.text
      ? [this._Description.text.text]
      : [];
    const subtitle = this._Subtitle.text.text ? [this._Subtitle.text.text] : [];
    const title = this._Title.text.text ? [this._Title.text.text] : [];
    return [...title, ...subtitle, ...description].join(',');
  }

  get _DataItem() {
    return this.tag('DataItem');
  }
  get _Content() {
    return this.tag('DataItem')._Content;
  }
  get _Title() {
    return this._Content.tag('Title');
  }
  get _Subtitle() {
    return this._Content.tag('Subtitle');
  }
  get _Description() {
    return this._Content.tag('Description');
  }
}
