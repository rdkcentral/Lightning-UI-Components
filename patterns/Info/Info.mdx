import { Canvas, Story } from '@storybook/addon-docs/blocks';
import Info from '.';

# Info

Generic pattern for Data Item specific to series/episode.

```js
import { Info } from '@lightning/ui/patterns';

Info: {
  type: Info,
  title: 'Series Title',
  subtitle: 'Season 5 - Info 2',
  description: 'This is a description of episode 2 from Season 5 of Series Title'
}
```

<Canvas>
  <Story id="patterns-info--basic" />
</Canvas>

### Properties

name|type|required|default|description
--|--|--|--|--
title|text|false|-|text to display in title
subtitle|text|false|-|text to display in subtitle
description|text|false|-|text to display in description
announce|text|false|title + subtitle + description|override default announce



