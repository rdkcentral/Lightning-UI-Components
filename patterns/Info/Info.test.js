import TestUtils from '../../test/lightning-test-utils';
import Info from '.';

const createInfo = TestUtils.makeCreateComponent(Info);

const title = 'Series Title';
const subtitle = 'Season 5 - Info 2';
const description =
  'This is a description of episode 2 from Season 5 of Series Title';

describe('Info', () => {
  let component, testRenderer;

  beforeEach(() => {
    [component, testRenderer] = createInfo({
      title,
      subtitle,
      description
    });
  });
  afterEach(() => {
    component = null;
    testRenderer = null;
  });

  it('renders', () => {
    const tree = testRenderer.toJSON(2);
    expect(tree).toMatchSnapshot();
  });

  it('set title', () => {
    expect(component._Title.text.text).toBe('Series Title');
  });

  it('set subtitle', () => {
    expect(component._Subtitle.text.text).toBe('Season 5 - Info 2');
  });

  it('set description', () => {
    expect(component._Description.text.text).toBe(
      'This is a description of episode 2 from Season 5 of Series Title'
    );
  });

  it('update title', () => {
    component.title = 'Update the title';
    expect(component._Title.text.text).toBe('Update the title');
  });

  it('update subtitle', () => {
    component.subtitle = 'Update the subtitle';
    expect(component._Subtitle.text.text).toBe('Update the subtitle');
  });

  it('update description', () => {
    component.description = 'Update the description';
    expect(component._Description.text.text).toBe('Update the description');
  });

  it('get announce text', () => {
    expect(component.announce).toBe(
      'Series Title,Season 5 - Info 2,This is a description of episode 2 from Season 5 of Series Title'
    );
  });

  it('update height', () => {
    component._focus();
    component._update();
    expect(component.h).toBe(
      component._DataItem.h *
        component.styles.getFocusScale(
          component._DataItem.w,
          component._DataItem.h
        ) +
        component._DataItem.y
    );
    component._unfocus();
    component._update();
    expect(component.h).toBe(component._DataItem.h + component._DataItem.y);
  });
});
