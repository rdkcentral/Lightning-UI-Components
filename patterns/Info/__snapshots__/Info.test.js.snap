// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders 1`] = `
Object {
  "Component": Object {
    "DataItem": Object {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": Object {
        "Background": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": "7a232328",
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 200,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Background",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Background",
          ],
          "texture": Object {
            "type": "RectangleTexture",
          },
          "visible": true,
          "w": 410,
          "x": 0,
          "y": 0,
          "zIndex": 2,
        },
        "Content": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": Object {
            "Description": Object {
              "active": false,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Description",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": Array [
                "Description",
              ],
              "texture": Object {
                "fontFace": "XfinityStandardMedium",
                "fontSize": 24,
                "lineHeight": 32,
                "text": "This is a description of episode 2 from Season 5 of Series Title",
                "textColor": 3439329279,
                "type": "TextTexture",
                "verticalAlign": "middle",
                "wordWrapWidth": 378,
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
            "Subtitle": Object {
              "active": false,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Subtitle",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": Array [
                "Subtitle",
              ],
              "texture": Object {
                "fontFace": "XfinityBrownBold",
                "fontSize": 28,
                "letterSpacing": -0.2,
                "lineHeight": 40,
                "text": "Season 5 - Info 2",
                "textColor": 4076863487,
                "type": "TextTexture",
                "verticalAlign": "middle",
                "wordWrapWidth": 378,
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
            "Title": Object {
              "active": false,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Title",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": Array [
                "Title",
              ],
              "texture": Object {
                "fontFace": "XfinityBrownBold",
                "fontSize": 36,
                "letterSpacing": -0.2,
                "lineHeight": 48,
                "text": "Series Title",
                "type": "TextTexture",
                "verticalAlign": "middle",
                "wordWrapWidth": 378,
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": true,
          "flexItem": true,
          "h": 168,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Content",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Content",
          ],
          "visible": true,
          "w": 378,
          "x": 16,
          "y": 16,
          "zIndex": 3,
        },
        "FocusRing": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": Object {
            "Ring": Object {
              "active": false,
              "alpha": 0.001,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": false,
              "h": 0,
              "isComponent": undefined,
              "mount": 0.5,
              "mountX": 0.5,
              "mountY": 0.5,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Ring",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": Array [
                "Ring",
              ],
              "texture": Object {
                "type": "StaticCanvasTexture",
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4125945074,
          "colorBl": "8aececf2",
          "colorBr": "14ffffff",
          "colorUl": "f5ececf2",
          "colorUr": "8aececf2",
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 200,
          "hasFinalFocus": false,
          "hasFocus": false,
          "isComponent": true,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "FocusRing",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": "",
          "tag": [Function],
          "tags": Array [
            "FocusRing",
          ],
          "type": "FocusRing",
          "visible": true,
          "w": 410,
          "x": 0,
          "y": 0,
          "zIndex": 1,
        },
      },
      "clipping": false,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 200,
      "hasFinalFocus": true,
      "hasFocus": true,
      "isComponent": true,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "DataItem",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": Array [
        "DataItem",
      ],
      "type": "DataItem",
      "visible": true,
      "w": 410,
      "x": 0,
      "y": 0,
      "zIndex": 0,
    },
  },
}
`;
